sdk/plugin/v5/plugin.go

```diff
@@ -0,0 +1,178 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/hashicorp/vault/sdk/helper/pluginutil"
+	"github.com/hashicorp/vault/sdk/logical"
+	bplugin "github.com/hashicorp/vault/sdk/plugin"
+	"github.com/hashicorp/vault/sdk/plugin/pb"
+)
+
+// BackendPluginClient is a wrapper around backendPluginClient
+// that also contains its plugin.Client instance. It's primarily
+// used to cleanly kill the client on Cleanup()
+type BackendPluginClient struct {
+	client pluginutil.PluginClient
+	sync.Mutex
+
+	logical.Backend
```

Comment:
```suggestion
	Backend logical.Backend
```

Reply:
If I name this field, I will have to provide forwarding methods for each of the functions defined in the logical.Backend interface. This is because we return a BackendPluginClient as a logical.Backend in [NewPluginClient](https://github.com/hashicorp/vault/pull/14946/files#diff-b93ddffb983c69704f7ca006ae541782e92daae082996f70d184d62bb2a0ea4bR51)

I can do that method forwarding. But it seems unnecessary when I can simply embed the interface to make BackendPluginClient a logical.Backend.

Reply:
I missed that, thanks for the explainer